meson.build

```diff
@@ -0,0 +1,83 @@
+project('euclidean-rhythms.lv2','c',version: '0.1.0-SNAPSHOT',default_options: ['default_library=shared','c_std=c99'])
+
+# Install folder
+lv2_directory = join_paths(get_option('libdir'), 'lv2')
+install_folder = join_paths(lv2_directory, meson.project_name())
+
+# Sources to compile
+euclidean_rhythms_src = ['plugins/eurhythms.c', 'src/euclidean_algorithms.c']
+
+# Dependencies for euclidean rhythms
+lv2_dep = dependency('lv2', required: true)
+m_dep = meson.get_compiler('c').find_library('m', required: true)
+all_dep = [lv2_dep,m_dep]
+
+# Get the host operating system and cpu architecture
+current_os = host_machine.system()
+current_arch = build_machine.cpu_family()
+
+# Shared c_args for libraries
+lib_c_args = ['-fvisibility=hidden']
+
+# Add default x86 and x86_64 optimizations
+if current_arch == 'x86' or current_arch == 'x86_64' and current_os != 'darwin'
+    lib_c_args += ['-msse','-msse2','-mfpmath=sse','-ffast-math','-fomit-frame-pointer','-fno-finite-math-only']
+endif
+
+# Configure extension for shared object
+if current_os == 'darwin' #mac
+    extension = '.dylib'
+else #unix like (linux, bsd)   
+    extension = '.so'
+endif 
+if current_os == 'windows' #windows
+    extension = '.dll'
+endif
+
+# Build of the shared object
+library('eurhythms',
```

Comment:
Since this must be a shared object, never a static one, you can use `shared_library(...)` to enforce that regardless of the project default for generic `library()`.

Although since this is a plugin, you might prefer `shared_module()`, the meson function for "this shared object is meant to be dlopened, and doesn't get things like a soname, and is allowed to have unresolved symbols that are provided by the executable that dlopens this module".

Reply:
Thanks for this!

Unfortunately, when I tried to use `shared_module`  instead of `shared_library`  I got a deprecation warning:
```
DEPRECATION: target test_euclidean links against shared module euclidean, which is incorrect.
             This will be an error in the future, so please use shared_library() for euclidean instead.
             If shared_module() was used for euclidean because it has references to undefined symbols,
             use shared_libary() with `override_options: ['b_lundef=false']` instead.
```
because the unit tests are linked against the library.